pkg/ui/console/client.go

```diff
@@ -164,6 +164,10 @@ func (c Client) DownloadProviders(ctx context.Context) (diags diag.Diagnostics)
 	checkUpdateOpts := core.CheckUpdatesOptions{
 		Providers: core.ManagedProviders(c.PluginManager, c.Providers),
 	}
+	if viper.GetBool("no-provider-update") {
```

Comment:
Why not move this on L162? That way it won't print the `Checking available provider updates...` message.